presto-parser/src/main/java/com/facebook/presto/sql/tree/LogicalBinaryExpression.java

```diff
@@ -82,6 +82,24 @@ public Expression getRight()
         return right;
     }
 
+    public boolean needsJoinWarning()
```

Comment:
The operator has public visibility, so this doesn't need to be a method on this class.  I'd put it as a helper method in `StatementAnalyzer`.